neusomatic/python/scan_alignments.py

```diff
@@ -79,30 +80,43 @@ def scan_alignments(work, scan_alignments_binary, input_bam,
 
     if not split_region_files:
         if regions_bed_file:
-            regions_bed = pybedtools.BedTool(
-                regions_bed_file).sort().merge(d=0)
+            regions_bed = bedtools_sort(regions_bed_file, run_logger=logger)
+            regions_bed = bedtools_merge(
+                regions_bed, args=" -d 0", run_logger=logger)
         else:
-            intervals = []
+            regions_bed = get_tmp_file()
             with pysam.AlignmentFile(input_bam, "rb") as samfile:
-                for chrom, length in zip(samfile.references, samfile.lengths):
-                    intervals.append(pybedtools.Interval(chrom, 1, length - 1))
-            regions_bed = pybedtools.BedTool(intervals).saveas()
+                with open(regions_bed, "w") as tmpfile:
+                    for chrom, length in zip(samfile.references, samfile.lengths):
+                        tmpfile.write("{}\t{}\t{}\t.\t.\t.\n".format(
+                            chrom, 1, length - 1))
         if not os.path.exists(work):
             os.mkdir(work)
-        total_len = sum(map(lambda x: int(x[2]) - int(x[1]) + 1, regions_bed))
-
+        total_len = 0
+        with open(regions_bed) as r_f:
+            for line in r_f:
+                if not line.strip():
+                    continue
+                chrom, st, en = line.strip().split("\t")[0:3]
+                total_len += int(en) - int(st) + 1
         if not restart:
             split_region_files = glob.glob(os.path.join(work, "region_*.bed"))
-            spilt_total_len = sum(map(lambda x: sum(
-                [y.length for y in pybedtools.BedTool(x)]), split_region_files))
+            spilt_total_len = 0
+            for split_file in split_region_files:
+                with open(split_file) as s_f:
+                    for line in s_f:
+                        if not line.strip():
+                            continue
+                        chrom, st, en = line.strip().split("\t")[0:3]
+                        spilt_total_len += int(en) - int(st)
             if spilt_total_len >= 0.98 * total_len:
```

Comment:
What is the 0.98 parameter for? @msahraeian

Reply:
@marghoob done.